packages/features/timezone-buddy/components/AvailabilitySliderTable.tsx

```diff
@@ -0,0 +1,165 @@
+                size="sm"
+                alt={username || email}
+                imageSrc={"/" + username + "/avatar.png"}
+                gravatarFallbackMd5="fallback"
+              />
+              <div className="">
+                <div
+                  className="text-emphasis max-w-64 truncate text-sm font-medium leading-none"
+                  title={email}>
+                  {username || "No username"}
+                </div>
+                <div className="text-subtle text-sm leading-none">{timeZone}</div>
+              </div>
+            </div>
+          );
+        },
+      },
+      {
+        id: "timezone",
+        accessorFn: (data) => data.timeZone,
+        header: "Timezone",
+        cell: ({ row }) => {
+          const { timeZone } = row.original;
+          const timeRaw = dayjs().tz(timeZone);
+          const time = timeRaw.format("HH:mm");
+          const utcOffsetInMinutes = timeRaw.utcOffset();
+          const hours = Math.abs(Math.floor(utcOffsetInMinutes / 60));
+          const minutes = Math.abs(utcOffsetInMinutes % 60);
+          const offsetFormatted = `${utcOffsetInMinutes < 0 ? "-" : "+"}${hours
+            .toString()
+            .padStart(2, "0")}:${minutes.toString().padStart(2, "0")}`;
+
+          return (
+            <div className="flex flex-col gap-2 leading-none">
+              <span className="text-default text-sm font-medium">{time}</span>
+              <span className="text-subtle text-xs ">GMT {offsetFormatted}</span>
+            </div>
+          );
+        },
+      },
+      {
+        id: "slider",
+        header: () => {
+          return (
+            <div className="space-between flex items-center space-x-4">
+              <span>{browsingDate.format("DD dddd MMM, YYYY")}</span>
+              <ButtonGroup>
+                <Button
+                  color="minimal"
+                  variant="icon"
+                  StartIcon={ChevronLeftIcon}
+                  onClick={() => setBrowsingDate(browsingDate.subtract(1, "day"))}
+                />
+                <Button
+                  onClick={() => setBrowsingDate(browsingDate.add(1, "day"))}
+                  color="minimal"
+                  StartIcon={ChevronRightIcon}
+                  variant="icon"
+                />
+              </ButtonGroup>
+            </div>
+          );
+        },
+        cell: ({ row }) => {
+          const { timeZone, dateRanges } = row.original;
+          // return <pre>{JSON.stringify(dateRanges, null, 2)}</pre>;
+          return <TimeDial timezone={timeZone} dateRanges={dateRanges} />;
+        },
+      },
+    ];
+
+    return cols;
+  }, [browsingDate]);
+
+  //we must flatten the array of arrays from the useInfiniteQuery hook
+  const flatData = useMemo(() => data?.pages?.flatMap((page) => page.rows) ?? [], [data]) as SliderUser[];
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+  const totalFetched = flatData.length;
```

Comment:
We can but it actually leads to more code -> Since we can't use memo or have an easy way to define variables inside there. You have to useState/let. This is the cleanest way i've found to do this rn. Will likely try to transform this whole pagination thing into a hook at some point in the future